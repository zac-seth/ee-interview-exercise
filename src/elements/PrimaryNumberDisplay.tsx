import { FC } from "react";

import { NumberDisplay, NumberDisplayMode } from "./NumberDisplay.style";

export type PrimaryNumberDisplayProps = {
  value: string;
};

export const PrimaryNumberDisplay: FC<PrimaryNumberDisplayProps> = ({
  value
}) => <NumberDisplay mode={NumberDisplayMode.PRIMARY}>{value}</NumberDisplay>;
