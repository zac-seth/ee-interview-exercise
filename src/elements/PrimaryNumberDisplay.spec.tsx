import { render, screen } from '@testing-library/react';

import { PrimaryNumberDisplay } from './PrimaryNumberDisplay';

describe(`PrimaryNumberDisplay`, () => {
  it(`Should render the component`, () => {
    render(<PrimaryNumberDisplay value='101' />);

    expect(screen.getByText(`101`)).toMatchSnapshot();
  });
});
