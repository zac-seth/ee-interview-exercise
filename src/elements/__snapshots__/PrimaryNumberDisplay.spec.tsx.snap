// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryNumberDisplay Should render the component 1`] = `
.c0 {
  text-shadow: 0 0 5px #FFF,0 0 15px #B6FF00;
  color: #FFF;
  margin: 0;
  padding: 0;
  font-size: 4rem;
  line-height: 4rem;
}

<p
  class="c0"
  mode="primary"
>
  101
</p>
`;
